platform_channels/lib/src/pet_list_screen.dart

```diff
@@ -23,16 +24,25 @@ class _PetListScreenState extends State<PetListScreen> {
     // to PetModel.
     BasicMessageChannel('stringCodecDemo', StringCodec())
         .setMessageHandler((message) async {
-      setState(() {
-        petListModel = PetListModel.fromJson(message);
-      });
+      if (message == null) {
+        scaffoldKey.currentState.showSnackBar(
```

Comment:
@Piinks, I'd like to call you in for an expert consultation here. :)

Is it possible/advisable to make sure the scaffold is available before making the call to show the snackbar? I would assume that if the scaffold's not in the tree that can only mean this widget (and its message channel) are also not in the tree, but if there's a best practice to use until your new messenger class lands, I'd love to show it.

Reply:
Since it's almost certainly an unreachable scenario and soon to be made obsolete by the arrival of ScaffoldMessenger, I'm going to retract my earlier statement and say we should just go with this.